pkg/sql/lease.go

```diff
@@ -1673,3 +1675,70 @@ func (m *LeaseManager) RefreshLeases(s *stop.Stopper, db *client.DB, g *gossip.G
 		}
 	})
 }
+
+// tableLeaseRefreshLimit is the upper-limit on the number of table leases
+// that will continuously have their lease refreshed.
+var tableLeaseRefreshLimit = settings.RegisterIntSetting(
+	"sql.tablecache.lease.refresh_limit",
+	"maximum number of tables to periodically refresh leases for",
+	50,
+)
+
+// PeriodicallyRefreshSomeLeases so that leases are fresh and can serve
+// traffic immediately.
+// TODO(vivek): Remove once epoch based table leases are implemented.
+func (m *LeaseManager) PeriodicallyRefreshSomeLeases() {
+	m.stopper.RunWorker(context.Background(), func(ctx context.Context) {
+		tickDuration := m.leaseDuration / 2
+		ticker := time.NewTicker(tickDuration)
+		tickChan := ticker.C
+		defer ticker.Stop()
+		for {
+			select {
+			case <-m.stopper.ShouldQuiesce():
+				return
+
+			case <-tickChan:
+				m.refreshSomeLeases(ctx)
+			}
+		}
+	})
+}
+
+// Refresh some of the current leases.
+func (m *LeaseManager) refreshSomeLeases(ctx context.Context) {
+	limit := tableLeaseRefreshLimit.Get(&m.execCfg.Settings.SV)
+	if limit <= 0 {
+		return
+	}
+	// Construct a list of tables needing their leases to be reacquired.
+	m.mu.Lock()
+	ids := make([]sqlbase.ID, 0, len(m.mu.tables))
+	var i int64
+	for k := range m.mu.tables {
```

Comment:
I think we want to see this with some initial IDs on boot to avoid the "cold start" problem -- though as-is, this should help with the "cooling off".

My prior PR iterated the descriptors table to get IDs: https://github.com/cockroachdb/cockroach/pull/27017/files

Reply:
I'm not too fond of that approach. Let's keep this and introduce a fix for cold start separately. Hold off on reviewing further until I have a test. Thanks!